Source/Backend/IPFTestRunnerConfiguration.h

```diff
@@ -8,11 +8,11 @@
 
 @interface IPFTestRunnerConfiguration : NSObject <NSCopying>
 
-@property (nonatomic, readonly, copy) NSString *hostname;
-@property (nonatomic, readonly) NSUInteger port;
-@property (nonatomic, readonly) NSUInteger duration;
-@property (nonatomic, readonly) NSUInteger streams;
-@property (nonatomic, readonly) IPFTestRunnerConfigurationType type;
+@property (nonatomic, copy) NSString *hostname;
+@property (nonatomic) NSUInteger port;
+@property (nonatomic) NSUInteger duration;
+@property (nonatomic) NSUInteger streams;
+@property (nonatomic) IPFTestRunnerConfigurationType type;
```

Comment:
These were made read-write so it can be easily configured from the view. The test runner class still gets a read only copy of it later when the test starts, so this change shouldn't be an issue. Let me know if you think otherwise

Reply:
These are generated remodel value files:
https://github.com/facebook/remodel

They are immutable but have builders so you can mutate or create them. It is a bit of a hassle, and probably overkill for this little app, but I like the concept, and they're easily serializable for persistence too!

If you'd like to make changes, I would modify `IPFTestRunnerConfiguration.value` and then run `make` in that folder (you will have to install `remodel` via npm first if I remember correctly).

Reply:
Remodel seems to only generate immutable structures. With the form mutating the structure now whenever one of the configurations is tweaked (in the closures like onTextChange and onSegmentChange), it needs to be mutable. Agree that immutable models reduce future bugs, but since this is a config which is then passed into Test Runner (And screen is not editable while a test is running), I don't think there is a huge value add in going the remodel way. If serialization is a consideration for future, we should convert this to swift and make it Codable, so it can be easily converted using a JSONEncoder or PropertyListEncoder for storage.